backend/api/views.py

```diff
@@ -111,4 +115,101 @@ class ImageViewSet(BaseViewSet):
     serializer = ImageSerializer
 
     def queryset_selector(self, request, wss):
-        return wss.images
\ No newline at end of file
+        return wss.images
+
+
+class ErrorResponse(Response):
+    
+    def __init__(self, data, status_code: int = 400):
+        super().__init__(data)
+        self.status_code = status_code
+
+
+class PaymentViewSet(viewsets.ViewSet):
+    authentication_classes = [SessionAuthentication, BasicAuthentication]
+    permission_classes = [IsAuthenticated]
+
+    @action(methods=['GET'], detail=False)
+    def request(self, request, year):
+        wss = get_wss_object_or_404(year)
+        if not wss.registration_open:
+            return ErrorResponse({
+                'message': "Sorry, the registration has been ended."
+            })
+        
+        callback_url = request.query_params.get("callback", None)
+
+        if callback_url is None:
+            return ErrorResponse({
+                'message': "`callback_url` should be passed in query string"
+            })
+        
+        user_profile = UserProfile.objects.get(user=request.user)
+        participant = Participant.objects.filter(current_wss=wss, user_profile=user_profile).first()
+
+        if participant is not None:
```

Comment:
Maybe its `payment_status` is not `OK` yet. Or is `payment_status` redundant at all?!

Reply:
I think that's redundant. So I'll remove it.